initial-ai-src/script.js

```diff
@@ -0,0 +1,587 @@
+class RuterDiscountCalculator {
+  constructor() {
+    this.maxDiscount = 40; // Maximum discount percentage
+
+    // Zone-based single ticket prices (NOK)
+    this.zonePrices = {
+      1: 39, // Zone 1 only
+      2: 55, // Zones 1-2
+      3: 73, // Zones 1-3
+      4: 91, // Zones 1-4
+      all: 109, // All zones
+    };
```

Comment:
🎨 All the values in this constructor are static, and never change. Move them out of the constructor and make them `consts`. Keep code as flat as possible: by nesting this code inside a class, inside a constructor - you've introduced 2 layers of necessary indent. This makes the code harder to read, especially for longer lines. (minor sidenote: it's probably worse for perf. Fine in this case, but maybe not if you keep writing code like this for more heavily-used classes)